### PR TITLE
sanity testing scenario 

### PR DESCRIPTION
**Pull request by dmitryAII**

### CODE REVIEW

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                      cluster_host_ip=manager1.private_ip_address,
+                      cluster_node_name=manager1.ip_address)
+
+    manager2.use()
+    cfy.cluster.join(manager1.ip_address,
+                     timeout=600,
+                     cluster_host_ip=manager2.private_ip_address,
+                     cluster_node_name=manager2.ip_address)
+
+    cfy.cluster.nodes.list()
```
Comment (tehasdf) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: after you create the cluster best wait for it to stabilize before returning (join finish)

like this https://github.com/cloudify-cosmo/cloudify-system-tests/pull/646/files#diff-413e5570ec61db36da808db62184dc15R61

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+from cosmo_tester.framework.examples.nodecellar import NodeCellarExample
+from cosmo_tester.framework.test_hosts import TestHosts
+from cosmo_tester.framework.util import prepare_and_get_test_tenant
+
+from cosmo_tester.test_suites.snapshots import (
+    create_snapshot,
+    download_snapshot,
+    upload_snapshot,
+    restore_snapshot,
+    # upgrade_agents,    # bug in agents upgrade
```
Comment (mcouthon) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: Put the Jira number here, so we'll know what to do when we fix it

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    logger.info('Creating secret agent_private_key_path as blueprint input')
+    cfy.secrets.create('agent_private_key_path', '-s',
+                       manager1.remote_private_key_path)
+
+    logger.info('Creating secret host_ip as blueprint input')
+    cfy.secrets.create('host_ip', '-s', manager1.ip_address)
+
+
+def _configure_tenant(cfy, manager, logger):
```
Comment (mcouthon) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: This function creates a user. Why is it called configure tenant?

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    cfy.users.create(USER_NAME, '-p', USER_PASS)
+
+    logger.info('Adding user to tenant')
+    cfy.tenants('add-user', USER_NAME, '-t', TENANT_NAME, '-r', 'user')
+
+
+def _set_sanity_user(cfy, manager, logger):
+    logger.info('Using manager `{0}`'.format(manager.ip_address))
+    cfy.profiles.set('-u', USER_NAME, '-p', USER_PASS, '-t', TENANT_NAME)
+    manager.use()
```
Comment (mcouthon) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: No need to `use` after `set`

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+
+def _set_admin_user(cfy, manager, logger):
+    logger.info('Using manager `{0}`'.format(manager.ip_address))
+    cfy.profiles.set('-u', 'admin', '-p', 'admin', '-t', 'default_tenant')
+    manager.use()
```
Comment (mcouthon) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: No need to `use` after `set`

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    create_snapshot(manager1, snapshot_id, attributes, logger)
+    download_snapshot(manager1, local_snapshot_path, snapshot_id, logger)
+
+    _set_admin_user(cfy, manager3, logger)
+
+    logger.info('Uploading and restoring snapshot')
+    upload_snapshot(manager3, local_snapshot_path, snapshot_id, logger)
+    restore_snapshot(manager3, snapshot_id, cfy, logger)
+    verify_services_status(manager3)
+    # wait a while to allow the restore-snapshot post-workflow commands to run
```
Comment (mcouthon) [inline cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: This sleep is actually for agents reconnection

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py (hunk 2 above)
Comment (dmitryAII) [reply cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: fixed

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py (hunk 3 above)
Comment (dmitryAII) [reply cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: fixed

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py (hunk 4 above)
Comment (dmitryAII) [reply cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: fixed

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py (hunk 5 above)
Comment (dmitryAII) [reply cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: fixed

File: cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py (hunk 6 above)
Comment (dmitryAII) [reply cosmo_tester/test_suites/bootstrap_based_tests/sanity_scenario_test.py]: fixed

**Pull request merged**